src/jit/codegencommon.cpp

```diff
@@ -8734,6 +8734,29 @@ void CodeGen::genFnEpilog(BasicBlock* block)
                     }
                     break;
 
+                case IAT_RELPVALUE:
+                {
+                    // Load the address into a register, load relative indirect and call through a register
+                    // We have to use R12 since we assume the argument registers are in use
+                    callType   = emitter::EC_INDIR_R;
+                    indCallReg = REG_R12;
+                    addr       = NULL;
+
+                    regNumber vptrReg1     = REG_R11;
+                    regMaskTP vptrReg1Mask = genRegMask(vptrReg1);
+                    inst_IV(INS_push, (int)vptrReg1Mask);
```

Comment:
There's a fundamental problem here, or maybe two: (1) you are temporarily trashing R11, which is the frame pointer. This will break stack walking, and local variable access (e.g., debugging), and if an exception occurs after FP has been trashed, FP wouldn't get restored, and the stack would also be off, and (2) we never change the stack pointer outside of the prolog or epilog. I think that's required, also for stack walking.

Presumably if you ran GCStress in a fully-interruptible method with this type of call, you would hit a problem (since GC would be attempted after the `push`). Actually, I'm guessing GC would fail dramatically because we don't track stack pointer moves outside the prolog/epilog, so the GC info would be incorrect.

The correct way to do this would be to make information known to the register allocator that we need a temporary register here, and then grab that temporary. This _might_ be a case where we've never done that, so I don't know if there's an established mechanism for communicating this.

I notice also that `StubLinkerCPU::ThumbEmitCallManagedMethod` has apparently the same call sequence, which uses R11 in the same way. I'm not sure what the requirements are on stubs, and whether this is ok in that context or not.

Reply:
@BruceForstall, what if we use some other register for this instead of R11? For example, R1?

Reply:
Those two things would probably make this work. However, it wouldn't be the "right" way to do it. The right way is to get the register allocator to allocate a temporary register that would be available at this point. It done, then it would automatically save/restore any conflicting register (if needed), and allocate stack space (if needed).

Reply:
@BruceForstall I've pushed new commits related to these. Could you, please, take a look?